sklearn/decomposition/tests/test_dict_learning.py

```diff
@@ -135,6 +145,25 @@ def test_dict_learning_online_shapes():
     assert_equal(np.dot(code, dictionary).shape, X.shape)
 
 
+def test_dict_learning_online_positivity():
+    rng = np.random.RandomState(0)
+    n_components = 8
+
+    dico = MiniBatchDictionaryLearning(
+        n_components, transform_algorithm='lasso_lars', random_state=0,
+        transform_positive=True, fit_positive=True).fit(X)
+    code = dico.transform(X)
+    assert_true((dico.components_ >= 0).all())
+    assert_true((code >= 0).all())
+
+    code, dictionary = dict_learning_online(X, n_components=n_components,
+                                            alpha=1, random_state=rng,
+                                            dict_positive=True,
+                                            code_positive=True)
+    assert_true((dictionary >= 0).all())
+    assert_true((code >= 0).all())
```

Comment:
And maybe also test for other combinations such as: (`dict_positive=True` and `dict_positive=False`) and (`dict_positive=False` and `dict_positive=True`).

`